packages/applicant/src/pages/submissions/[submissionId]/equality-and-diversity/[grantBeneficiaryId]/organisation.page.tsx

```diff
@@ -0,0 +1,152 @@
+import {
+  postGrantBeneficiaryResponse,
+} from '../../../../../services/GrantBeneficiaryService';
+import callServiceMethod from '../../../../../utils/callServiceMethod';
+import { getJwtFromCookies } from '../../../../../utils/jwt';
+import {
+  errorPageParams,
+  errorPageRedirect,
+} from '../equality-and-diversity-service-errors';
+import { EqualityAndDiversityParams } from '../types';
+import { fetchGrantBeneficiary } from './fetchGrantBeneficiary';
+
+export enum OrganisationRadioOptions {
+  VCSE = 'Voluntary, community and social enterprise',
+  SME = 'Small to medium enterprise',
+  NEITHER = 'Neither',
+}
+
+type RequestBody = {
+  organisation: `${OrganisationRadioOptions}`;
+};
+
+export const getServerSideProps: GetServerSideProps<{}, EqualityAndDiversityParams> = async ({
+  params,
+  resolvedUrl,
+  req,
+  res,
+  query,
+}) => {
+  const { submissionId, grantBeneficiaryId } = params as Record<string, string>;
+  const { returnToSummaryPage } = query as Record<string, string>;
+  let defaultChecked = null;
+
+  let grantBeneficiary: GrantBeneficiary;
+  try {
+    grantBeneficiary = await fetchGrantBeneficiary(submissionId, req);
+  } catch (_) {
+    return errorPageRedirect(submissionId);
+  }
+
+  if (grantBeneficiary.organisationGroup1) defaultChecked = OrganisationRadioOptions.VCSE;
+  else if (grantBeneficiary.organisationGroup2) defaultChecked = OrganisationRadioOptions.SME;
+  else if (grantBeneficiary.organisationGroup3) defaultChecked = OrganisationRadioOptions.NEITHER;
+
+  const response = await callServiceMethod(
+    req,
+    res,
+    async (body: RequestBody) => {
+      if (body.organisation) {
+        await postGrantBeneficiaryResponse(
+          {
+            submissionId: submissionId,
+            hasProvidedAdditionalAnswers: true,
+            organisationGroup1:
+              body.organisation === OrganisationRadioOptions.VCSE,
+            organisationGroup2:
+              body.organisation === OrganisationRadioOptions.SME,
+            organisationGroup3: body.organisation === OrganisationRadioOptions.NEITHER,
+          },
+          getJwtFromCookies(req),
+          grantBeneficiaryId
+        );
+      }
+    },
+    returnToSummaryPage
+      ? `/submissions/${submissionId}/equality-and-diversity/${grantBeneficiaryId}/summary`
+      : `/submissions/${submissionId}/equality-and-diversity/${grantBeneficiaryId}/sex`,
+    errorPageParams(submissionId)
+  );
+
+  if ('redirect' in response) {
+    return response;
+  }
+
+  const { publicRuntimeConfig } = getConfig();
+
+  return {
+    props: {
+      formAction: `${publicRuntimeConfig.subPath}${resolvedUrl}`,
+      skipURL: `${
+        publicRuntimeConfig.subPath
+      }/submissions/${submissionId}/equality-and-diversity/${grantBeneficiaryId}/${
+        returnToSummaryPage ? 'summary' : 'sex'
+      }`,
+      defaultChecked: defaultChecked,
+      csrfToken: (req as any).csrfToken?.() || '',
+    },
+  };
+};
+
```

Comment:
Can we use a more specific type here eg:

```suggestion
      defaultChecked,
      csrfToken: (req as Req).csrfToken?.() || '',
    },
  };
};

type Req = IncomingMessage & {
  cookies: Partial<{
    [key: string]: string;
  }>;
} & { csrfToken: () => string };

```